packages/story-editor/src/components/panels/design/layer/group.js

```diff
@@ -0,0 +1,512 @@
+    duplicateGroupById: actions.duplicateGroupById,
+    groupLayers: state.currentPage.elements.filter(
+      (el) => el.groupId === groupId
+    ),
+  }));
+
+  const group = groups[groupId];
+  const groupRef = useRef(null);
+  const deleteButtonRef = useRef(null);
+  const groupDomId = `group-${groupId}`;
+
+  const lockTitle = group.isLocked
+    ? __('Unlock Group', 'web-stories')
+    : __('Lock Group', 'web-stories');
+
+  const LockIcon = group.isLocked ? Icons.LockClosed : Icons.LockOpen;
+
+  const layerName = group.name;
+  const [newLayerName, setNewLayerName] = useState(layerName);
+  const { isSelected, handleClick } = useGroupSelection(groupId);
+  const isLayerNamingEnabled = useFeature('layerNaming');
+  const { renamableLayer, setRenamableLayer } = useCanvas(
+    ({ state, actions }) => ({
+      renamableLayer: state.renamableLayer,
+      setRenamableLayer: actions.setRenamableLayer,
+    })
+  );
+  const isRenameable = renamableLayer?.elementId === groupId;
+
+  const handleChange = (evt) => {
+    setNewLayerName(evt.target.value);
+  };
+
+  const handleKeyDown = (evt) => {
+    if (evt.key === 'Escape') {
+      setNewLayerName(layerName);
+      setRenamableLayer(null);
+    }
+  };
+
+  const updateLayerName = () => {
+    setRenamableLayer(null);
+    const trimmedLayerName = newLayerName.trim();
+    // Don't update name if trimmed layer name is empty.
+    // This means that submitting an empty name will exit renaming, and the
+    // layer name will revert to whatever it was before, ignoring the empty input.
+    if (!trimmedLayerName) {
+      setNewLayerName(layerName);
+    } else {
+      updateGroupById({
+        groupId,
+        properties: {
+          name: trimmedLayerName,
+        },
+      });
+    }
+  };
+
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+    updateLayerName();
+  };
+
+  // We need to prevent the pointer-down event from propagating to the
+  // reorderable when you click on the input. If not, the reorderable will
+  // move focus, which will blur the input, which will cancel renaming.
+  const stopPropagation = (evt) => evt.stopPropagation();
+
+  const handleLockGroup = () => {
+    updateGroupById({
+      groupId,
+      properties: { isLocked: !group.isLocked },
+    });
+    updateElementsById({
+      elementIds: groupLayers.map((layer) => layer.id),
+      properties: { isLocked: !group.isLocked },
+    });
+  };
+
+  const handleDuplicateGroup = () => {
+    const newGroupId = uuidv4();
+    const newGroupName = generateGroupName(groups, group.name);
+    duplicateGroupById({
+      oldGroupId: groupId,
+      groupId: newGroupId,
+      name: newGroupName,
+    });
+  };
+
+  return (
+    <LayerContainer>
+      {isRenameable && isLayerNamingEnabled ? (
+        <LayerInputWrapper>
+          <GroupIconsWrapper>
+            <Icons.ChevronDown />
+            <Icons.Group />
+          </GroupIconsWrapper>
```

Comment:
The layout here is a bit tight. The group name needs some spacing to the icon, IMHO:

<img width="268" alt="Screen Shot 2022-06-13 at 12 31 58" src="https://user-images.githubusercontent.com/637548/173335213-c641385e-1457-41d5-8f62-3c36fb596b79.png">